src/Classes/TradeQueryRateLimiter.lua

```diff
@@ -243,14 +246,23 @@ function TradeQueryRateLimiterClass:AgeOutRequests(policy, time)
             for window, windowValue in pairs(rule.state) do
                 if timestamp >= (requestHistory.lastCheck - window) and timestamp < (now - window) then
                     -- timestamp that used to be in the window on last check
-                    windowValue.request = math.max(windowValue.request - 1, 0)
+                    if not windowValue.decremented then
+                        windowValue.request = math.max(windowValue.request - 1, 0)
+                        windowValue.decremented = true
+                    end
                 end
             end
         end
         if timestamp < now - requestHistory.maxWindow then
             table.remove(requestHistory.timestamps, i)
         end
     end
+    -- Reset flags after processing
+	for _, rule in pairs(self.policies[policy]) do
+		for window, windowValue in pairs(rule.state) do
+			windowValue.decremented = nil
+		end
+	end
```

Comment:
Hi, these lines should be space-indented.  (And no, I don't know why tests use spaces! 🙃)